plugins/inputs/x509_cert/x509_cert.go

```diff
@@ -101,52 +106,75 @@ func (c *X509Cert) Gather(acc telegraf.Accumulator) error {
+				fields := getFields(cert, now)
+				tags := getTags(cert, location.String())
+
+				// The first certificate is the leaf/end-entity certificate which
+				// needs DNS name validation against the URL hostname.
+				opts := x509.VerifyOptions{
+					Intermediates: x509.NewCertPool(),
+					KeyUsages:     []x509.ExtKeyUsage{x509.ExtKeyUsageAny},
+					Roots:         c.tlsCfg.RootCAs,
+					DNSName:       dnsName,
+				}
+				// Reset DNS name to only use it for the leaf node
+				dnsName = ""
+
+				// Add all returned certs to the pool if intermediates except for
+				// the leaf node and ourself
+				for j, c := range certs.PeerCertificates[1:] {
+					if i+1 != j {
+						opts.Intermediates.AddCert(c)
+					}
 				}
-			}
 
-			if _, err := cert.Verify(opts); err == nil {
-				tags["verification"] = "valid"
-				fields["verification_code"] = 0
-			} else {
-				c.Log.Debugf("Invalid certificate at index %2d!", i)
-				c.Log.Debugf("  cert DNS names:    %v", cert.DNSNames)
-				c.Log.Debugf("  cert IP addresses: %v", cert.IPAddresses)
-				c.Log.Debugf("  cert subject:      %v", cert.Subject)
-				c.Log.Debugf("  cert issuer:       %v", cert.Issuer)
-				c.Log.Debugf("  opts.DNSName:      %v", opts.DNSName)
-				c.Log.Debugf("  verify options:    %v", opts)
-				c.Log.Debugf("  verify error:      %v", err)
-				c.Log.Debugf("  location:          %v", location)
-				c.Log.Debugf("  tlsCfg.ServerName: %v", c.tlsCfg.ServerName)
-				c.Log.Debugf("  ServerName:        %v", c.ServerName)
-				tags["verification"] = "invalid"
-				fields["verification_code"] = 1
-				fields["verification_error"] = err.Error()
-			}
+				if _, err := cert.Verify(opts); err == nil {
+					tags["verification"] = "valid"
+					fields["verification_code"] = 0
+				} else {
+					c.Log.Debugf("Invalid certificate at index %2d!", i)
+					c.Log.Debugf("  cert DNS names:    %v", cert.DNSNames)
+					c.Log.Debugf("  cert IP addresses: %v", cert.IPAddresses)
+					c.Log.Debugf("  cert subject:      %v", cert.Subject)
+					c.Log.Debugf("  cert issuer:       %v", cert.Issuer)
+					c.Log.Debugf("  opts.DNSName:      %v", opts.DNSName)
+					c.Log.Debugf("  verify options:    %v", opts)
+					c.Log.Debugf("  verify error:      %v", err)
+					c.Log.Debugf("  location:          %v", location)
+					c.Log.Debugf("  tlsCfg.ServerName: %v", c.tlsCfg.ServerName)
+					c.Log.Debugf("  ServerName:        %v", c.ServerName)
+					tags["verification"] = "invalid"
+					fields["verification_code"] = 1
+					fields["verification_error"] = err.Error()
+				}
+				// OCSPResponse only for leaf cert
+				if i == 0 && len(certs.OCSPResponse) > 0 {
+					resp, err := ocsp.ParseResponse(certs.OCSPResponse, nil)
+					if err != nil {
+						tags["ocsp_stapled"] = "no"
+						fields["ocsp_error"] = err.Error()
+					} else {
+						tags["ocsp_stapled"] = "yes"
+						// resp.Status: 0=Good 1=Revoked 2=Unknown
+						fields["ocsp_status"] = resp.Status
```

Comment:
Can we please make this a tag with the "human-readable" text?

Reply:
Like verification=valid tag and verification_code=0i field.
Would tag: ocsp_status=good field: ocsp_status_code=0i be ok ?
(I'd like to keep field, I think it'll probably make alert rules easier (at least with prometheus)).

Reply:
Yes that's what I had in mind.